lib/EnsEMBL/REST/Model/ga4gh/Beacon.pm

```diff
@@ -705,17 +743,55 @@ sub get_dataset_allele_response {
       $externalURL = "http://grch37.ensembl.org";
     }
 
-    if($sv == 1 && defined $vf) {
-      $externalURL .= "/Homo_sapiens/StructuralVariation/Explore?sv=" . $vf->variation_name();
-    }
-    elsif($sv == 0 && defined $vf) {
-      $externalURL .= "/Homo_sapiens/Variation/Explore?v=" . $vf->name();
+    my @urls;
+
+    if($vf) {
+      my $var_name;
+      my $delimiter;
+      foreach my $variant (@{$vf}) {
+        if($sv == 1) {
+          $var_name = $variant->variation_name();
+          $delimiter = "StructuralVariation/Explore?sv=";
+        }
+        else {
+          $var_name = $variant->name();
+          $delimiter = "Variation/Explore?v=";
+        }
+
+        # Checks if dataset_id is one of the datasets where variant is found
+        # If it's not found then dataset response won't include this variant
+        my %var_dataset = %{$variant_dt};
+        my $datasets = $var_dataset{$var_name};
+        my $contains = contains_value($dataset_id, $datasets);
+        if($contains) {
+          my $url_tmp = $externalURL . "/Homo_sapiens/" . $delimiter . $var_name;
+          push @urls, $url_tmp;
+        }
+      }
+      my $url = join(',', @urls);
+      $externalURL = $url;
     }
+
+    $ds_response->{'variantCount'} = scalar @urls;
+
     $ds_response->{'externalUrl'} = $externalURL;
 
   return $ds_response;
 }
 
+sub contains_value {
+  my ($value, $array) = @_;
+
+  my $related;
```

Comment:
This is now redundant and you can `return;`

Reply:
I missed it. Thanks!